packages/flutter/lib/src/widgets/editable_text.dart

```diff
@@ -747,8 +747,20 @@ class EditableTextState extends State<EditableText> with AutomaticKeepAliveClien
 
   // Calculate the new scroll offset so the cursor remains visible.
   double _getScrollOffsetForCaret(Rect caretRect) {
-    final double caretStart = _isMultiline ? caretRect.top : caretRect.left;
-    final double caretEnd = _isMultiline ? caretRect.bottom : caretRect.right;
+    double caretStart;
+    double caretEnd;
+    if (_isMultiline) {
+      caretStart = caretRect.top;
```

Comment:
I think should be caretRect.top - caretOffset

Reply:
I tried this and an integration test caught the fact that it will sometimes go below 0.0.  I think I'll try doing your suggestion but only when `caretRect.top != 0.0`.